chore!: don't add fileinfo columns to loaded files

dkrako opened the pull request:
## :boom: Breaking Change

This PR removes the adding of fileinfo columns (parsed from the filename using `filename_pattern`) to `Gaze` and `Events`. These columns always consisted of a single constant value per `Gaze`/`Events` object and were not of any use.

## Implemented changes

- [x] remove adding fileinfo columns when loading dataset `Gaze` files
- [x] remove adding fileinfo columns when loading dataset `Events` files
- [x] remove adding fileinfo columns when detecting events

## How Has This Been Tested?

This PR doesn not add any new tests.

- [x] adjusted existing tests
- [x] removed irrelevant add_fileinfo tests

## Context

#### required by:
- #1371

#### requires:
- [x] #1377

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

cbueth reviewed (changes requested):
Thank you for the PR! I do think it looks good, just want you to consider a few thoughts few points:

- Is this drop in `save_events` still needed?

https://github.com/pymovements/pymovements/blob/9a191bfccd1ba7d923ab7ae65bb27318e2d4b98b/src/pymovements/dataset/dataset_files.py#L602-L604

- This PR should not get rid of the whole `fileinfo` dataframe, right? There are still references in docstrings, and documentation/notebooks that might be relevant.

- We might want to rename these in `io.py`, `fileinfo_key` -> `column_key` for consistency - or in #1371:

https://github.com/pymovements/pymovements/blob/f06ecf692cb8c4e482d18a9b9887f1d85ae15741/src/pymovements/gaze/io.py#L505-L508

  There are two more of these list comprehensions in the file.

- How about adding regression tests checking that `Gaze.samples.columns` and `Events.columns` do not contain any fileinfo-derived fields after loading and detection?

dkrako commented:
> Thank you for the PR! I do think it looks good, just want you to consider a few thoughts few points:
> 
>     * Is this drop in `save_events` still needed?

Good catch, removed the redundancy.

>     * This PR should not get rid of the whole `fileinfo` dataframe, right? There are still references in docstrings, and documentation/notebooks that might be relevant.

Yes, the `fileinfo` dataframe is still kept until we manage to add a good replacement. The references in the docs "fortunately" didn't document this adding of fileinfo columns to the gaze and events objects so no updates necessary there.

>     * We might want to rename these in `io.py`, `fileinfo_key` -> `column_key` for consistency - or in #1371:

Good catch, changed to `column_key`/`column_dtype`.

> There are two more of these list comprehensions in the file.
> 
>     * How about adding regression tests checking that `Gaze.samples.columns` and `Events.columns` do not contain any fileinfo-derived fields after loading and detection?

As this isn't a bugfix I don't think it's worth to add and keep the tests for all future versions.
